src/ngResource/resource.js

```diff
@@ -721,7 +718,13 @@ angular.module('ngResource', ['ng']).
                 (success || noop)(value, response.headers);
                 return value;
               },
-              responseErrorInterceptor);
+              responseErrorInterceptor || error ?
+              function(response) {
+                (error || noop)(response);
+                (responseErrorInterceptor || noop)(response);
+                return response;
```

Comment:
Isn't this changing the current behavior ?
I think we should either return the return value of `responseErrorInterceptor` or `$q.reject(response)`.

Or maybe something like:

```js
promise = promise.then(
  function(response) {
    var value = responseInterceptor(response);
    (success || noop)(value, response.headers);
    return value;
  }, function(response) {
    (error || noop)(response);
    return $q.reject(response);
  }).catch(responseErrorInterceptor);
```